### ISSUE TITLE
User Story 23, Adding Item Quantity to Cart

### ISSUE THREAD

**Issue by AngelaGuardia** [Arique1104/monster_shop_2005#23]
As a visitor
When I have items in my cart
And I visit my cart
Next to each item in my cart
I see a button or link to increment the count of items I want to purchase
I cannot increment the count beyond the item's inventory size

**Comment by Arique1104** [comment]
This sounds like I will need the following:

- a spec within the show page for cart
- some times to put into said cart
- three model methods:
1. One that increments the total number of items in the cart.
2. One that increments the number of items for that particular item.
3. One that checks these numbers don't surpass the total inventory.

**Comment by Arique1104** [comment]
Turns out I didn't need to do any model testing for this one.  I didn't make any model tests at all.  Our orders are being stored in a hash where the keys are the item_ids and the values are an integer representing the amount.

What I have NOT checked for is that you can NOT add an item that is outside of the inventory availability.

**Comment by Arique1104** [comment]
I'm confused as to whether I'll be able to reuse some other codes that exists and just apply some if statements.

As I'm looking at the add_more_item method, some of the logic that I want to exist in there has to access @contents.

I'm too tired to figure this out this second, but I wanted to jot some notes for myself tomorrow.  I have to finish this early and I didn't get as far as I would've liked this evening.

Basically, I have to make some logic within the Cart class that I can utilize in the Cart Controller class, and then make sure that I circle back so that I add my tests in my models to make sure the calculations are working as I expect.

**Comment by Arique1104** [comment]
I'm really stuck on this.  I've added some logic on my `add_more_item` method, but I've realized that I can't just rely solely on the `item.inventory` because:
- There are no officially orders that will make the total inventory = 0.

So in order for me to have this casually check to make sure there aren't more items added that surpass inventory I have to develop a new number that actually tracks that and makes sure that my cart doesn't have more items then there are inventory during a pre-order phase.
